src/search/engines/duckduckgo/duckduckgo.go

```diff
@@ -50,17 +50,25 @@ func Search(ctx context.Context, query string, relay *bucket.Relay, options engi
 				} else {
 					linkScheme = "http://"
 				}
-				titleText = strings.TrimSpace(row.Find(dompaths.Title).Text())
+				titleText = _sedefaults.SanitizeTitle(row.Find(dompaths.Title).Text())
 			case 1:
-				descText = strings.TrimSpace(row.Find(dompaths.Description).Text())
+				descText = _sedefaults.SanitizeDescription(row.Find(dompaths.Description).Text())
 			case 2:
 				rawURL := linkScheme + row.Find("td > span.link-text").Text()
-				linkText = parse.ParseURL(rawURL)
+				linkText = _sedefaults.SanitizeURL(rawURL)
 			case 3:
-				if hrefExists && linkText != "" && linkText != "#" && titleText != "" {
-					res := bucket.MakeSEResult(linkText, titleText, descText, Info.Name, page, (i/4 + 1))
-					bucket.AddSEResult(&res, Info.Name, relay, options, pagesCol)
+				if !hrefExists {
+					log.Error().
+						Str("engine", Info.Name.String()).
+						Str("url", linkText).
+						Str("title", titleText).
+						Str("description", descText).
+						Msgf("duckduckgo.Search(): href attribute doesn't exist on matched URL element (%v)", dompaths.Link)
+					return
 				}
+
+				res := bucket.MakeSEResult(linkText, titleText, descText, Info.Name, page, (i/4 + 1))
+				bucket.AddSEResult(&res, Info.Name, relay, &options, pagesCol)
```

Comment:
Missing returned bool check (valid := and if valid)

Reply:
What should happen if its invalid?

Reply:
Oh right there's no need for `pageRankCounter[pageIndex]++`. My bad.